patroni/postgresql.py

```diff
@@ -622,12 +626,12 @@ def restart(self):
 
     def _write_postgresql_conf(self):
         # rename the original configuration if it is necessary
-        if not os.path.exists(self._postgresql_base_conf):
+        if not self._postgresql_custom_conf and not os.path.exists(self._postgresql_base_conf):
```

Comment:
This prevents a `postgresql.base.conf` to be created from an original `postgresql.conf` if there is a custom conf somewhere else, as it will not be used.

The following lines will overwrite the `postgresql.conf` on the data dir anyway.